ts/parts/Legend.ts

```diff
@@ -561,12 +561,21 @@ class Legend {
             legendGroup = item.legendGroup;
 
         if (legendGroup && legendGroup.element) {
-            legendGroup[defined(legendGroup.translateY) ? 'animate' : 'attr']({
+            const shouldAnimate = defined(legendGroup.translateY);
+
+            legendGroup[shouldAnimate ? 'animate' : 'attr']({
                 translateX: ltr ?
                     itemX :
                     legend.legendWidth - itemX - 2 * (symbolPadding as any) - 4,
                 translateY: itemY
             });
+
```

Comment:
I think (without testing it) that it may be both safer and cheaper to use the `complete` callback for animation. Can you check that? Something like this:

```js
            const complete = (): void => {
                fireEvent(this, 'afterPositionItem', { item });
            };
            const attribs = {
                translateX: ltr ?
                    itemX :
                    legend.legendWidth - itemX - 2 * (symbolPadding as any) - 4,
                translateY: itemY
            };

            if (defined(legendGroup.translateY)) {
                legendGroup.animate(attribs, { complete });
            } else {
                legendGroup.attr(attribs);
                complete();
            }
```
I haven't counted the bytes though :) But I assume under any circumstance it is better practice to use the `complete` callback than a timeout.